src/sentry/static/sentry/app/views/projectDetail/projectLatestReleases.tsx

```diff
@@ -0,0 +1,195 @@
+    return [
+      ['releases', `/projects/${organization.slug}/${projectSlug}/releases/`, {query}],
+    ];
+  }
+
+  /**
+   * If our releases are empty, determine if we had a release in the last 90 days (empty message differs then)
+   */
+  async onLoadAllEndpointsSuccess() {
+    const {releases} = this.state;
+    const {organization, projectSlug} = this.props;
+
+    if ((releases ?? []).length !== 0) {
+      this.setState({hasOlderReleases: true});
+      return;
+    }
+
+    this.setState({loading: true});
+
+    const oldReleases = await this.api.requestPromise(
+      `/projects/${organization.slug}/${projectSlug}/releases/`,
+      {
+        method: 'GET',
+        query: {
+          statsPeriod: '90d',
+          per_page: 1,
+        },
+      }
+    );
+
+    this.setState({hasOlderReleases: oldReleases.length > 0, loading: false});
+  }
+
+  handleTourAdvance = (index: number) => {
+    const {organization, projectId} = this.props;
+
+    analytics('releases.landing_card_clicked', {
+      org_id: parseInt(organization.id, 10),
+      project_id: projectId && parseInt(projectId, 10),
+      step_id: index,
+      step_title: RELEASES_TOUR_STEPS[index].title,
+    });
+  };
+
+  renderReleaseRow = (release: Release) => {
+    const {projectId} = this.props;
+    const {lastDeploy, dateCreated} = release;
+
+    return (
+      <React.Fragment>
+        <DateTime date={lastDeploy?.dateFinished || dateCreated} seconds={false} />
+        <VersionWrapper>
+          <Version
+            version={release.version}
+            tooltipRawVersion
+            projectId={projectId}
+            truncate
+          />
+        </VersionWrapper>
+      </React.Fragment>
+    );
+  };
+
+  renderInnerBody() {
+    const {loading, releases, hasOlderReleases} = this.state;
+    const checkingForOlderReleases =
+      (releases ?? []).length === 0 && hasOlderReleases === undefined ? true : false;
+    const showLoadingIndicator = loading || checkingForOlderReleases;
+
+    if (showLoadingIndicator) {
+      return <Placeholder height="160px" />;
+    }
+
+    if (!hasOlderReleases) {
+      return (
+        <div>
+          <StyledButtonBar gap={1}>
+            <Button size="small" priority="primary" external href={DOCS_URL}>
+              {t('Start Setup')}
+            </Button>
+            <FeatureTourModal
+              steps={RELEASES_TOUR_STEPS}
+              onAdvance={this.handleTourAdvance}
+            >
+              {({showModal}) => (
+                <Button size="small" onClick={showModal}>
+                  {t('Get a tour')}
+                </Button>
+              )}
+            </FeatureTourModal>
+          </StyledButtonBar>
+        </div>
+      );
+    }
+
+    if ((releases ?? []).length === 0) {
+      return t('No releases match the filter.');
+    }
+
+    return <ReleasesTable>{(releases ?? []).map(this.renderReleaseRow)}</ReleasesTable>;
```

Comment:
at this point `releases` should be `defined` and have the length higher than `0`, should we still add `?? [] ` ? maybe we can verify if it is `null` in the condition above